frontend/components/Documents.js

```diff
@@ -17,6 +23,67 @@ const Documents = () => {
             });
     }, []);
 
+    const handleTitleInputChange = (event) => {
+	    event.persist();
+	    setNewDocData((values) => ({
+		    ...values,
+		    title: event.target.value
+	    }));
+    };
+
+    const handleDateInputChange = (event) => {
+	    event.persist();
+	    setNewDocData((values) => ({
+		    ...values,
+		    date: event.target.value
+	    }));
+    };
+
+    const handleTextInputChange = (event) => {
+	    event.persist();
+	    setNewDocData((values) => ({
+		    ...values,
+		    text: event.target.value
+	    }));
+    };
+
+    const handleAuthorInputChange = (event) => {
+	    event.persist();
+	    setNewDocData((values) => ({
+		    ...values,
+		    author: event.target.value
+	    }));
+    };
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        const csrftoken = getCookie("csrftoken");
```

Comment:
We should probably perform some kind of validation here to ensure that all of the required fields are accounted for. As is stands, submitting with filling out all the fields will cause the API to return a 500 status code.

Reply:
I've added validation for when `text` is empty and updated the error that appears when a user doesn't input a date.